src/Features/Core/Portable/CodeRefactoringHelpers.cs

```diff
@@ -131,5 +217,38 @@ internal static class CodeRefactoringHelpers
 
             return TextSpan.FromBounds(start, end);
         }
+
+        /// <summary>
+        /// Strips leading and trailing whitespace from <paramref name="span"/>.
+        /// </summary>
+        /// <remarks>
+        /// Returns unchanged <paramref name="span"/> in case <see cref="TextSpan.IsEmpty"/>.
+        /// Returns empty Span with original <see cref="TextSpan.Start"/> in case it contains only whitespace.
+        /// </remarks>
+        private static async Task<TextSpan> GetStrippedTextSpan(Document document, TextSpan span, CancellationToken cancellationToken)
+        {
+            if (span.IsEmpty)
+            {
+                return span;
+            }
+
+            var sourceText = await document.GetTextAsync(cancellationToken).ConfigureAwait(false);
+            var start = span.Start;
+            var end = span.End;
+
+            while (start < end && char.IsWhiteSpace(sourceText[end - 1]))
+            {
+                end--;
+            }
+
+            while (start < end && char.IsWhiteSpace(sourceText[start]))
+            {
+                start++;
+            }
+
+            return start == end
+                ? new TextSpan(start, 0)
+                : TextSpan.FromBounds(start, end);
+        }
```

Comment:
i don't know why it can't always be TextSpan.FromBounds.  I looked a the code of it, and there should be no problem if end and start are the same.  All it does is this:

```c#
            if (start < 0)
            {
                throw new ArgumentOutOfRangeException(nameof(start), CodeAnalysisResources.StartMustNotBeNegative);
            }

            if (end < start)
            {
                throw new ArgumentOutOfRangeException(nameof(end), CodeAnalysisResources.EndMustNotBeLessThanStart);
            }

            return new TextSpan(start, end - start);
```

Reply:
You're right. It's weird, had some crashes once just skimmed the code later but it was probably some other issue. Will fix (even though that's more previous PR issue).